### PR TITLE
Remove gradients on buttons

### PR DESCRIPTION
**Pull request by rollerozxa**
This PR removes the default gradient on buttons:

image

## To do
This PR is Ready for Review.

## How to test
Look.

### CODE REVIEW

File: src/gui/guiSkin.cpp [hunk 1]
```diff
@@ -24,95 +24,47 @@ GUISkin::GUISkin(EGUI_SKIN_TYPE type, video::IVideoDriver* driver)
...
-		Sizes[EGDS_TEXT_DISTANCE_Y] = 2;
-
-		Sizes[EGDS_TITLEBARTEXT_DISTANCE_X] = 3;
-		Sizes[EGDS_TITLEBARTEXT_DISTANCE_Y] = 2;
-	}
+	Colors[EGDC_3D_DARK_SHADOW]     = video::SColor(101,50,50,50);
+	Colors[EGDC_3D_SHADOW]          = video::SColor(101,130,130,130);
+	Colors[EGDC_3D_FACE]            = video::SColor(220,70,70,70);
+	Colors[EGDC_3D_HIGH_LIGHT]      = video::SColor(101,255,255,255);
+	Colors[EGDC_3D_LIGHT]           = video::SColor(101,210,210,210);
```
Comment (sfan5) [inline src/gui/guiSkin.cpp:31]: the faux 3d thing should probably be disabled too?

**Comment by SmallJoker** [comment]
I prefer the current buttons. Others may prefer flat ones. A general solution would IMO be to use 9slice drawing for all buttons so that games and players could overwrite the textures (normal, hover, press) on demand.

**Comment by Desour** [comment]
Tested. Somehow looks better in-game than in screenshots.
One small issue though: The arrows in scrollbars are a little low contrast.

I'm fine with using this in mainmenu.

For mods/subgames though, I'd prefer to have some sort of styling option for this. I don't care much about the default (IIRC, we wanted to make the default ugly anyway). But users (=server owners) should have the possibility to reproduce the classic experience.

**Comment by Zughy** [comment]
>  we wanted to make the default ugly anyway

That's a terrible approach. If I open the engine for the first time, I shouldn't be pushed to run away and never open it again. Default should be pretty, not ugly

**Comment by Desour** [comment]
> > we wanted to make the default ugly anyway
> 
> That's a terrible approach. If I open the engine for the first time, I shouldn't be pushed to run away and never open it again. Default should be pretty, not ugly

If you open the game, you see the main menu, not the defaults for mods.
(Btw. here's where that suggestion came from: <https://irc.minetest.net/minetest-dev/2023-07-14#i_6096922>)

**Comment by rubenwardy** [comment]
> (Btw. here's where that suggestion came from: [irc.minetest.net/minetest-dev/2023-07-14#i_6096922](https://irc.minetest.net/minetest-dev/2023-07-14#i_6096922))

To be clear, I was joking

**Comment by tigercoding56** [comment]
you know , why not just add a way to configure the default looks of buttons and stuff like that , call it themes or something ,

File: src/gui/guiSkin.cpp (hunk 1 above)
Comment (Zughy) [reply src/gui/guiSkin.cpp:31]: @rollerozxa

Comment (rollerozxa) [reply src/gui/guiSkin.cpp:31]: shouldn't that be kept such that the buttons still look "clickable"?

Comment (sfan5) [reply src/gui/guiSkin.cpp:31]: dunno, a comparison image would help judging this I think

Comment (Zughy) [reply src/gui/guiSkin.cpp:31]: @rollerozxa reminder

**Review by grorp — COMMENTED** [review]
I like this PR. Removing the gradients on buttons makes the mainmenu (and also the in-game GUIs) look fresher and less dusty in my eyes. I think keeping the "faux 3D thing", as this PR currently does, is a good idea.

My concerns:

- As a side effect, this PR also changes the color of mainmenu tabs. I think the old color is better, as it has a higher contrast to the background.

- > The arrows in scrollbars are a little low contrast. ~ Desour

- This PR makes groups of buttons which are very close to each other harder to recognize as separate buttons, for example here:
   
   screenshot of a group of buttons which are very close to each other

   The gradients helped in that regard.

**Comment by Zughy** [comment]
About the last point: they could be either separated more or put one next to the other without any space left

**Comment by rollerozxa** [comment]
> The arrows in scrollbars are a little low contrast.

As mentioned in a comment in #13909, it's possible to change the colour of these symbols. Would the symbols being white be better?

image

**Comment by grorp** [comment]
> Would the symbols being white be better?

Yes, that would be better.

**Comment by rollerozxa** [comment]
Also I can't remember if I intentionally changed the background colour of tabs, it's most likely a side effect of changing the 3D face colour to make the non-gradiented buttons darker.

**Comment by Bituvo** [comment]
I think this behavior could be made toggleable somehow. What if I *want* gradients on my buttons?

**Comment by Zughy** [comment]
@Bituvo custom main menu

**Comment by Montandalar** [comment]
Not personally a fan but it would put Minetest a bit closer to current design trends. The loss of gradients seems like a loss, will that remain easy to do with mod code even if the engine styling is changed?

The scrollbars *definitely* need white arrows; there's way too little contrast otherwise. The scrollbar background should be a bit lighter too, or rather maybe the scrollbar colour made a bit lighter as I will propose below.

I find the text a bit more readable with the gradients; I think the brightness of the new buttons is on the low side. I know that makes it a higher contrast compared to a lighter shade, but reading white text on black is harder for quite a few people with astigmatism. The white colours on a dark background blur much worse with astigmatism than black text does on a white background. Therefore, overall colour value not being too dark also matters, not pure contrast ratio. There is room for less colour contrast while still remaining inside WCAG accessibility guidelines. It seems like the buton it was chosen off the bottom of the gradient instead of nearer the middle.

I think the fact that there's no contrast between buttons and background is a problem. Can the main background shade be kept the same while the button shades are made lighter? The main menu looks terrible to me now:
roller-btn-remove-grad-mainmenu

With my proposal for lighter buttons, it would look more like this ("Sound Volume under cursor):
roller-btn-remove-grad-mainmenu-my

(I had to fill with the paint bucket in GIMP because I couldn't work out what colour is what in guiSkin.cpp in a short period; there's alpha values used so I couldn't just pick it with the colour picker tool either).

**Comment by appgurueu** [comment]
Which "action/change" is still needed? Is it [this](https://github.com/minetest/minetest/pull/13718#discussion_r1287692341)?

**Comment by Zughy** [comment]
@appgurueu yes, but please feel free to review this PR so we can all move on

**Comment by rubenwardy** [comment]
Honestly, I'm not too bothered by this change as games should be providing their own button styling and we're planning on redesigning the mainmenu in the future, which would come with its own button design

An alternative to this PR would be to just change the mainmenu buttons, and leave game buttons as is

**Comment by Zughy** [comment]
We've been stalling on a simple gradiente change for half a year now, c'mon...

**Comment by SmallJoker** [comment]
I do not like the proposed color palette of the PR. The 3D buttons provided a nice contrast on various backgrounds, hence the colors proposed in https://github.com/minetest/minetest/pull/13718#issuecomment-1848860128 might work better in some cases - but generally I would prefer to have this controlled by formspec prepends rather than making the contrast worse.

**Comment by grorp** [comment]
> I do not like the proposed color palette of the PR. The 3D buttons provided a nice contrast on various backgrounds, hence the colors proposed in https://github.com/luanti-org/luanti/pull/13718#issuecomment-1848860128 might work better in some cases - but generally I would prefer to have this controlled by formspec prepends to allow contrast fine-tuning depending on where the button is used.
> 
> As how it currently is: 👎 from my side.

@SmallJoker

<details>
<summary>Different brightnesses for <code>EGDC_3D_FACE</code> (alpha value is always 220, ignore alpha values in chat messages) in the MTG pause menu for comparison</summary>
<p>(100, 100, 100)<br />
screenshot</p>
<p>(90, 90, 90)<br />
screenshot</p>
<p>(80, 80, 80)<br />
screenshot</p>

<p>(75, 75, 75)<br />
screenshot</p>

<p>(70, 70, 70) (current version of this PR)<br />
screenshot</p>
<p>(60, 60, 60)<br />
screenshot</p>
</details>

(75, 75, 75) would about match the proposal in https://github.com/luanti-org/luanti/pull/13718#issuecomment-1848860128

This is what it would look like in the mainmenu:
screenshot

This seems ok to me, what do you think?

(The slightly brighter color also helps with the problems I mentioned in https://github.com/luanti-org/luanti/pull/13718#pullrequestreview-1690518737)

Regarding the "formspec prepend" concern: games/mods can already use 9-slice images and style[] to make their buttons look exactly the way they want, so we're only changing the default style here.

**Comment by SmallJoker** [comment]
> Different brightnesses for `EGDC_3D_FACE` (alpha value is always 220, ignore alpha values in chat messages) in the MTG pause menu for comparison
> This seems okay to me, would it also be okay from your side?

Thank you for the comparisons. The additional contrast with the background using bgcolor=`(220, 90, 90, 90)` does indeed help a bit. Unfortunately even after changing that, the buttons still look like text fields or boxes instead of something that can be interacted with. This topic is subjective ("aero"-like vs "material") and in my perception it's worse than the 3D effect that we're currently using.


On as side note. I noticed that there's a rather new button style in the main menu, thus I wondered how it might look like if we'd adapt it to the entire main menu... (to feed the topic with some alternative ideas)

9-slice
`tab_online.lua` modification:
```Lua
	local retval =
		"style_type[button,image_button;border=false;bgimg=" .. defaulttexturedir .. "/mybuttons.png;bgimg_middle=2,2,-2,-3]" ..
		-- Search
```

**Pull request closed without merging**